Package request: deno

Deno is a secure TypeScript runtime with V8 engine. Written in rust by the same creator as Node.js.

Recently version 1.0.0 was released so it's starting to become more stable.

GitHub: https://github.com/denoland/deno

Indeed

```
error[E0308]: mismatched types                                                             --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1ecc6299db9ec823/rusty_v8-0.16.0/build.rs:148:18                                                             |                                                                                   148 | fn platform() -> &'static str {                                                       |    --------      ^^^^^^^^^^^^ expected `&str`, found `()`                             |    |                                                                                  |    implicitly returns `()` as its body has no tail or `return` expression         
error: aborting due to previous error                                                                                                                                           For more information about this error, try `rustc --explain E0308`.                     error: could not compile `rusty_v8`
```

@stupidestmodder if got this error
```
error[E0308]: mismatched types
   --> build.rs:157:18
    |
157 | fn platform() -> &'static str {
    |    --------      ^^^^^^^^^^^^ expected `&str`, found `()`
    |    |
    |    implicitly returns `()` as its body has no tail or `return` expression

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rusty_v8`
```
you can go modify the build.rs in ../.cargo/registry/src/github.com-1ecc6299db9ec823/v8-0.34.0/build.rs
and find this
```
fn platform() -> &'static str {
  #[cfg(target_os = "windows")]
  {
    "win"
  }
  #[cfg(target_os = "linux")]
  {
    "linux64"
  }
  #[cfg(target_os = "macos")]
  {
    "mac"
  }
}
```
change to
```
fn platform() -> &'static str {
  #[cfg(target_os = "windows")]
  {
    "win"
  }
  #[cfg(any(target_os = "linux", target_os= "android"))]
  {
    "linux64"
  }
  #[cfg(target_os = "macos")]
  {
    "mac"
  }
}
```

--- a/packages/deno/build.sh
+++ b/packages/deno/build.sh
@@ -1,0 +1,31 @@
+TERMUX_PKG_HOMEPAGE=https://deno.land/
+TERMUX_PKG_DESCRIPTION="A modern runtime for JavaScript and TypeScript"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1.17.3
+TERMUX_PKG_SRCURL=https://github.com/denoland/deno.git
+TERMUX_PKG_DEPENDS="libffi"
+TERMUX_PKG_BUILD_DEPENDS="librusty-v8"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+# Due to dependency on librusty-v8.
+TERMUX_PKG_BLACKLISTED_ARCHES="arm, i686, x86_64"
+
+termux_step_make() {
+	termux_setup_rust
+	local libdir=target/$CARGO_TARGET_NAME/release/deps
+	mkdir -p $libdir
+	ln -sf $TERMUX_PREFIX/lib/libffi.so $libdir/
+	local libgcc="$($CC -print-libgcc-file-name)"
+	echo "INPUT($libgcc -l:libunwind.a)" > $libdir/libgcc.so
+	local cmd="cargo build --jobs $TERMUX_MAKE_PROCESSES \
+		--target $CARGO_TARGET_NAME --release"
+	$cmd || :
+	ln -sf $TERMUX_PREFIX/lib/librusty_v8.a \
+		target/$CARGO_TARGET_NAME/release/gn_out/obj/librusty_v8.a
+	$cmd
+}
+
+termux_step_make_install() {
+	install -Dm700 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/deno
+}

--- a/packages/deno/ext-ffi-Cargo.toml.patch
+++ b/packages/deno/ext-ffi-Cargo.toml.patch
@@ -1,0 +1,11 @@
+--- a/ext/ffi/Cargo.toml
++++ b/ext/ffi/Cargo.toml
+@@ -16,7 +16,7 @@
+ [dependencies]
+ deno_core = { version = "0.114.0", path = "../../core" }
+ dlopen = "0.1.8"
+-libffi = "2.0.0"
++libffi = { version = "2.0.0", features = ["system"] }
+ serde = { version = "1.0.129", features = ["derive"] }
+ tokio = { version = "1.10.1", features = ["full"] }
+ 

--- a/packages/deno/runtime-ops-signal.rs.patch
+++ b/packages/deno/runtime-ops-signal.rs.patch
@@ -1,0 +1,11 @@
+--- a/runtime/ops/signal.rs
++++ b/runtime/ops/signal.rs
+@@ -97,7 +97,7 @@
+   }
+ }
+ 
+-#[cfg(target_os = "linux")]
++#[cfg(any(target_os = "linux", target_os = "android"))]
+ pub fn signal_str_to_int(s: &str) -> Result<libc::c_int, AnyError> {
+   match s {
+     "SIGHUP" => Ok(1),

